aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -77,6 +77,30 @@ impl<T: Hash + Clone + Debug + Eq + Serialize, V: TransactionWrite> Default
 }
 
 impl<T: Hash + Clone + Debug + Eq + Serialize, V: TransactionWrite> VersionedGroupValue<T, V> {
+    fn set_base_values(
+        &mut self,
+        shifted_idx: ShiftedTxnIndex,
+        incarnation: Incarnation,
+        // TODO add layout to values
+        values: impl IntoIterator<Item = (T, V)>,
+    ) {
+        match self.idx_to_update.get(&shifted_idx) {
+            Some(previous) => {
+                // base value may have already been provided due to a concurrency race.
+                // If maybe_layout is None, they are required to be identical.
+                // If maybe_layout is Some, there might have been an exchange
+                // Assert the length of bytes for efficiency (instead of full equality)
+                for (tag, v) in values.into_iter() {
+                    let prev_v = previous
+                        .get(&tag)
+                        .expect("Reading twice from storage must be consistent");
+                    assert!(v.bytes_len() == prev_v.bytes_len());
+                }
+            },
+            None => self.write(shifted_idx, incarnation, values),
```

Comment:
What is the problem with overwriting concurrently the base values? They should be identical?

Reply:
not if there were aggregators, and they got exchanged with ephemeral IDs